### ISSUE TITLE
GAMS with Ipopt fail

### ISSUE THREAD

**Issue by andreaslundell** [coin-or/SHOT#135]
If Ipopt or Ipopth is selected as the NLP solver in GAMS, the NLP call fails with:

```
*** gevCallSolver: Solve failed (ipopth,RMINLP,sl=5)
*** gevCallSolver: rc=2 msg=ipoReadyAPI failed
```

For example:

```
./SHOT ~/Research/minlplib/gms/tls4.gms --mip=cbc Subsolver.GAMS.NLP.Solver=ipopth --nlp=gams
```
gives:

```
╶ Supporting Hyperplane Optimization Toolkit (SHOT) ──────────────────────────────────────────────────────────────────╴

 Andreas Lundell and Jan Kronqvist, Åbo Akademi University, Finland.
 See documentation for full list of contributors and utilized software libraries.

 Version: 1.1.0. Git hash: d77d9bab. Released: Aug 20 2021.

 For more information visit https://shotsolver.dev

╶ Modeling system ────────────────────────────────────────────────────────────────────────────────────────────────────╴

 Modeling system:            GAMS
 Problem read from file:     /home/andreas/Research/minlplib/gms/tls4.gms

 Time to extract information on quadratics: 0.000000
 Performing bound tightening on original problem.
  - Bounds for 18 variables tightened in 0.00 s and 2 passes.
 Performing bound tightening on reformulated problem.
  - Bounds for 8 variables tightened in 0.00 s and 2 passes.

╶ Problem instance ───────────────────────────────────────────────────────────────────────────────────────────────────╴

                                    Original             Reformulated

 Problem classification:            MINLP, nonconvex     MINLP, convex

 Objective function direction:      minimize             minimize
 Objective function type:           linear               linear

 Number of constraints:             64                   72
  - linear:                         60                   60
  - convex nonlinear:               0                    12
  - nonconvex nonlinear:            4                    0

 Number of variables:               105                  113
  - real:                           16                   24
  - binary:                         85                   85
  - integer:                        4                    4
  - nonlinear:                      20                   20

 Number of transformations performed:                    8
  - signomial terms partitioning:                        8

╶ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────────╴

 No options file specified.

 Options specified:

  - Dual.MIP.Solver = 2
  - Dual.TreeStrategy = 0
  - Model.Reformulation.Quadratics.Strategy = 0
  - Primal.FixedInteger.Solver = 1
  - Primal.Tolerance.TrustLinearConstraintValues = false
  - Subsolver.GAMS.NLP.Solver = ipopth

 Dual strategy:              Multi-tree
  - cut algorithm:           ESH
  - solver:                  Cbc 2.10

 Primal NLP solver:          IPOPTH in GAMS 36.1


╶ Interior point search ──────────────────────────────────────────────────────────────────────────────────────────────╴

 Strategy selected:          cutting plane minimax

    Iteration     │  Time  │    Cuts     │     Objective value     │  Objective diff.
     #: type      │  tot.  │   + | tot.  │    problem | line srch  │    abs. | rel.
╶─────────────────┴────────┴─────────────┴─────────────────────────┴──────────────────╴
     1: LP           0.05                      -1e+12 | -1e+12          inf. | inf.
     2: LP           0.05      3 | 3              -24 | 29.8059      5.4e+01 | 2.2e+00
     3: LP           0.05      3 | 6           -12.78 | 17.8589      3.1e+01 | 2.4e+00
     4: LP           0.06      3 | 9         -9.43449 | 8.2712       1.8e+01 | 1.9e+00
     5: LP           0.06      3 | 12          -7.973 | 3.41603      1.1e+01 | 1.4e+00
     6: LP           0.07      3 | 15        -7.77878 | 2.98475      1.1e+01 | 1.4e+00
     7: LP           0.07      3 | 18        -7.76863 | 2.7315       1.1e+01 | 1.4e+00
     8: LP           0.08      3 | 21        -7.74951 | 0.440303     8.2e+00 | 1.1e+00
     9: LP           0.08      3 | 24        -7.72154 | -0.126149    7.6e+00 | 9.8e-01

 Valid interior point with constraint deviation -0.126 found.

╶ Main iteration step ────────────────────────────────────────────────────────────────────────────────────────────────╴

    Iteration     │  Time  │  Dual cuts  │     Objective value     │   Objective gap   │     Current solution
     #: type      │  tot.  │   + | tot.  │       dual | primal     │    abs. | rel.    │    obj.fn. | max.err.
╶─────────────────┴────────┴─────────────┴─────────────────────────┴───────────────────┴──────────────────────────────╴

     1: LP-O         0.08                           0 | inf.            inf. | 1.0e+00            0 | 63: 6.34e+01
     4: LP-O         0.09      5 | 16        0.677373 | inf.            inf. | 1.0e+00     0.677373 | 62: 7.42e+00
     5: LP-O         0.10      6 | 22         1.30029 | inf.            inf. | 1.0e+00      1.30029 | 61: 3.47e+00
     6: LP-O         0.10      6 | 28         1.59129 | inf.            inf. | 1.0e+00      1.59129 | 64: 3.82e+00
     7: LP-O         0.11      4 | 32         1.64324 | inf.            inf. | 1.0e+00      1.64324 | 65: 1.73e+00
     8: LP-O         0.12      5 | 37         1.68228 | inf.            inf. | 1.0e+00      1.68228 | 60: 2.38e+00
    11: LP-O         0.13      4 | 48         1.68228 | inf.            inf. | 1.0e+00      1.68228 | 62: 2.89e-01
    12: MILP-SL1     0.16      5 | 53         1.74656 | inf.            inf. | 1.0e+00          6.1 | 68: 1.30e+01
*** gevCallSolver: Solve failed (ipopth,RMINLP,sl=5)
*** gevCallSolver: rc=2 msg=ipoReadyAPI failed
terminate called after throwing an instance of 'std::logic_error'
  what():  Calling GAMS NLP solver failed: Solve failed (ipopth,RMINLP,sl=5)
[1]    31037 abort      ./SHOT ~/Research/minlplib/gms/tls4.gms --mip=cbc  --nlp=gams
```

Can you verify @svigerske?

**Comment by svigerske** [comment]
When I try to reproduce on my system, the problem I get is that GAMS/Ipopt does not use the Ipopt library that comes with GAMS, but the one that was loaded by SHOT before.

Somehow also Cbc gets mixed in here, but `gams/gams/libipocclib64.so` should not call into `coin/install/lib/libipopt.so.3`, but `gams/gams/libipopt64.so`:
```
    13: MILP-SL1     5.76      6 | 63         3.49003 | inf.            inf. | 1.0e+00         10.5 | 65: 6.42e+00     
      |  
      | COIN-OR Ipopt    36.1.0 r2c0a44a Released Aug  2, 2021 LEG x86 64bit/Linux     
      | 
COIN-OR Interior Point Optimizer (Ipopt Library 3.14.1) 
      | written by A. Waechter, commercially supported by GAMS Development Corp. 
      | 
### diverging_iterates_tol (Real Number)  ###
Category: Termination
Description: Threshold for maximal value of primal iterates.
ÿ 
      |  
      |  < 
      | (1e+20 
      |  <= 
      | +inf 
terminate called after throwing an instance of 'Ipopt::OPTION_INVALID'


ERROR while running Cbc. Signal SIGABRT caught. Getting stack trace.
/home/stefan/work/coin/install/lib/libCbcSolver.so.0(+0x62d5d6) [0x7f46f68ad5d6]
/usr/lib/libc.so.6(+0x3cda0) [0x7f46f2259da0]
/usr/lib/libc.so.6(gsignal+0x142) [0x7f46f2259d22]
/usr/lib/libc.so.6(abort+0x116) [0x7f46f2243862]
/usr/lib/libstdc++.so.6(+0x99802) [0x7f46f25e3802]
/usr/lib/libstdc++.so.6(+0xa5c8a) [0x7f46f25efc8a]
/usr/lib/libstdc++.so.6(+0xa5cf7) [0x7f46f25efcf7]
/usr/lib/libstdc++.so.6(+0xa5f8e) [0x7f46f25eff8e]
/home/stefan/work/coin/install/lib/libipopt.so.3(_ZNK5Ipopt11OptionsList14GetStringValueERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERS6_S8_+0xf28) [0x7f46f324a57a]
/home/stefan/work/coin/install/lib/libipopt.so.3(_ZNK5Ipopt11OptionsList12GetBoolValueERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERbS8_+0x1dd) [0x7f46f324be71]
/home/stefan/work/gams/gams/libipocclib64.so(+0x2a090) [0x7f46bc3a3090]
/home/stefan/work/gams/gams/libipocclib64.so(ipoReadyAPI+0x9) [0x7f46bc3a37f0]
/home/stefan/work/gams/gams/libjoatdclib64.so(_Z42GEVDOORG_tgmsenvironment_DOT_gevcallsolverP29GEVDOORG_tgmsenvironment_OD_SPvPKhS3_iiS3_S3_diiddPS1_Ph+0x1239) [0x7f46c1209d89]
/home/stefan/work/gams/gams/libjoatdclib64.so(cgevcallsolver+0xf4) [0x7f46c11c49d4]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT13NLPSolverGAMS20solveProblemInstanceEv+0x361) [0x7f46f90672f5]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT13NLPSolverBase12solveProblemEv+0x4d) [0x7f46f9068225]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT33TaskSelectPrimalCandidatesFromNLP13solveFixedNLPEv+0x121d) [0x7f46f78676f1]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT33TaskSelectPrimalCandidatesFromNLP3runEv+0x1bc) [0x7f46f7866270]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT25SolutionStrategyMultiTree12solveProblemEv+0xd9) [0x7f46f781b08d]
/home/stefan/work/gams/shot/SHOT-build/libSHOTSolver.so(_ZN4SHOT6Solver12solveProblemEv+0x3d8) [0x7f46f752d9be]
./SHOT(+0x264e39) [0x55b27c340e39]
/usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f46f2244b25]
./SHOT(+0x25e59e) [0x55b27c33a59e]
```

**Comment by andreaslundell** [comment]
Closed with #137
